### PR TITLE
Improve simulator

### PR DESCRIPTION
**Pull request by EmilieNumworks**
- add haptic effects on smartphones
- highlight the selected key on all simulators

### CODE REVIEW

File: ion/src/device/shared/events.cpp [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#include <ion/events.h>
+
+namespace Ion {
+namespace Events {
+
+void didPressNewKey() {
```
Comment (EmilieNumworks) [inline ion/src/device/shared/events.cpp:6]: Check that no jump instruction is emitted for the device

Comment (EmilieNumworks) [reply ion/src/device/shared/events.cpp:6]: Checked!

File: ion/src/simulator/android/src/cpp/haptics.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  }
+}
+
+void shutdown() {
+  if (sSDLHaptic) {
+    SDL_HapticClose(sSDLHaptic);
+  }
+}
+
+void perform() {
```
Comment (Ecco) [inline ion/src/simulator/android/src/cpp/haptics.cpp:28]: `void rumble();`

File: ion/src/simulator/android/src/cpp/haptics.cpp [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    }
+  }
+}
+
+void shutdown() {
+  if (sSDLHaptic) {
+    SDL_HapticClose(sSDLHaptic);
+  }
+}
+
```
Comment (Ecco) [inline ion/src/simulator/android/src/cpp/haptics.cpp:27]: add a `bool isEnabled()` that is re-implemented per platform. The dummy should default to false as a failsafe (add a comment).

**Pull request merged**